src/main/java/se/sundsvall/partyassets/api/validation/impl/ValidStatusReasonConstraintValidator.java

```diff
@@ -15,36 +23,63 @@ abstract class ValidStatusReasonConstraintValidator {
 
 	private static final String ERROR_MESSAGE_TEMPLATE = "'%s' is not valid reason for status %s. Valid reasons are %s.";
 
+	private static final String MUNICIPALITY_ID = "municipalityId";
+
 	private final StatusService statusService;
 
-	protected ValidStatusReasonConstraintValidator(StatusService statusService) {
+	protected ValidStatusReasonConstraintValidator(final StatusService statusService) {
 		this.statusService = statusService;
 	}
 
 	boolean noStatusReason(final Status status, final String statusReason) {
 		if (allNull(status, statusReason)) {
 			return true;
 		}
+		final var municipalityId = getPathVariable(MUNICIPALITY_ID);
+
+		return !statusService.getReasonsForAllStatuses(municipalityId).containsKey(status) && isNull(statusReason);
+	}
 
-		return !statusService.getReasonsForAllStatuses().containsKey(status) && isNull(statusReason);
+	/** TODO: Remove this and use the DEPT-44 version instead when that is implemented (UF-8954)
+	 * Getting value for path variable name from current request
+	 *
+	 * @param variableName path parameter to receive value for
+	 * @return value of path parameter that matches sent in variable name
+	 */
+	String getPathVariable(final String variableName) {
+		return Stream.ofNullable(RequestContextHolder.getRequestAttributes())
+			.map(req -> req.getAttribute(URI_TEMPLATE_VARIABLES_ATTRIBUTE, SCOPE_REQUEST))
+			.filter(Objects::nonNull)
+			.filter(Map.class::isInstance)
+			.map(Map.class::cast)
+			.map(map -> map.get(variableName))
+			.filter(Objects::nonNull)
+			.filter(String.class::isInstance)
+			.map(String.class::cast)
+			.findAny()
+			.orElse("2281");
```

Comment:
Should the method default to 2281 or should it throw exception if municpalityId is not found? 

If method should return 2281, remove inparameter and rename method to getMunicipalityIdFromPath (or likely) as it will always return 2281 if variableName is not found (no matter what value that is sent to the method).